test/eventlistener_test.go

```diff
@@ -302,21 +305,51 @@ func TestEventListenerCreate(t *testing.T) {
 	// ElPort forward sink pod for http request
 	portString := strconv.Itoa(*eventReconciler.ElPort)
 	podName := sinkPods.Items[0].Name
-	cmd := exec.Command("kubectl", "port-forward", podName, "-n", namespace, fmt.Sprintf("%s:%s", portString, portString))
-	err = cmd.Start()
-	if err != nil {
-		t.Fatalf("Error starting port-forward command: %s", err)
-	}
-	if cmd.Process == nil {
-		t.Fatalf("Error starting command. Process is nil")
-	}
+	stopChan, errChan := make(chan struct{}, 1), make(chan error, 1)
+
 	defer func() {
-		if err = cmd.Process.Kill(); err != nil {
-			t.Fatalf("Error killing port-forward process: %s", err)
-		}
+		close(stopChan)
 	}()
-	// Wait for port forward to take effect
-	time.Sleep(5 * time.Second)
+	go func(stopChan chan struct{}, errChan chan error) {
+		readyChan := make(chan struct{}, 1)
+		config, err := clientcmd.BuildConfigFromFlags("", knativetest.Flags.Kubeconfig)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		roundTripper, upgrader, err := spdy.RoundTripperFor(config)
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", namespace, podName)
+		hostIP := strings.TrimLeft(config.Host, "htps:/")
```

Comment:
is the `htps:/` supposed to be `https:/`?

Reply:
It seem TrimLeft just gets a set of characters and remove all the ones from the beginning of the string. I can use the other function that only removes the exact keyword if you prefer.

Reply:
yeah, I would use `strings.TrimPrefix` instead, or call `url.Parse` and then set the path afterwards.